src/EnergyPlus/FanCoilUnits.cc

```diff
@@ -2095,28 +2105,39 @@ namespace FanCoilUnits {
-									FanCoil( FanCoilNum ).MaxIterIndexH );
-							}
-						} else if ( SolFlag == -2 ) {
-							++FanCoil( FanCoilNum ).LimitErrCountH;
-							if ( FanCoil( FanCoilNum ).LimitErrCountH < 2 ) {
-								ShowWarningError( "Hot Water control failed in fan coil unit " + FanCoil( FanCoilNum ).Name );
-								ShowContinueError( "  Bad hot water mass flow limits" );
-								ShowContinueErrorTimeStamp( "..Water flow rate set to lower limit " );
-							} else {
-								ShowRecurringWarningErrorAtEnd( "Hot Water control failed in fan coil unit " + FanCoil( FanCoilNum ).Name,
-									FanCoil( FanCoilNum ).BadMassFlowLimIndexH );
+							// tighten limits on water flow rate to see if this allows convergence
+							CoolingLoad = false;
+							HeatingLoad = true;
+							TightenWaterFlowLimits( FanCoilNum, CoolingLoad, HeatingLoad, FanCoil( FanCoilNum ).HotControlNode, ControlledZoneNum, FirstHVACIteration, QZnReq, MinWaterFlow, MaxWaterFlow );
+							SolveRegulaFalsi( 0.001, MaxIterCycl, SolFlag, HWFlow, CalcFanCoilHWLoadResidual, MinWaterFlow, MaxWaterFlow, Par );
+							if ( SolFlag == -1 ) {
+								++FanCoil( FanCoilNum ).ConvgErrCountH;
+								if ( FanCoil( FanCoilNum ).ConvgErrCountH < 2 ) {
+									ShowWarningError( "Hot Water control failed in fan coil unit " + FanCoil( FanCoilNum ).Name );
+									ShowContinueError( "  Iteration limit exceeded in calculating water flow rate " );
+									Node( FanCoil( FanCoilNum ).HotControlNode ).MassFlowRate = HWFlow;
+									Calc4PipeFanCoil( FanCoilNum, ControlledZoneNum, FirstHVACIteration, QUnitOut );
+									ShowContinueErrorTimeStamp( "Load Request = " + TrimSigDigits( QZnReq ) + ", Final Capacity = " + TrimSigDigits( QUnitOut ) );
+									ShowContinueErrorTimeStamp( "Min water flow used during iterations = " + TrimSigDigits( MinWaterFlow ) + ", Max water flow used during iterations = " + TrimSigDigits( MaxWaterFlow ) );
+									ShowContinueErrorTimeStamp( "Water flow rate on last iteration = " + TrimSigDigits( HWFlow ) );
+									ShowContinueErrorTimeStamp( "..Water flow rate set to last iteration value " );
+								} else {
+									ShowRecurringWarningErrorAtEnd( "Hot water flow Iteration limit exceeded in fan coil unit " + FanCoil( FanCoilNum ).Name,
+										FanCoil( FanCoilNum ).MaxIterIndexH );
+								}
+							} else if ( SolFlag == -2 ) {
+								++FanCoil( FanCoilNum ).LimitErrCountH;
+								if ( FanCoil( FanCoilNum ).LimitErrCountH < 2 ) {
+									ShowWarningError( "Hot Water control failed in fan coil unit " + FanCoil( FanCoilNum ).Name );
+									ShowContinueError( "  Bad hot water mass flow limits" );
+									ShowContinueErrorTimeStamp( "..Water flow rate set to lower limit " );
+								} else {
+									ShowRecurringWarningErrorAtEnd( "Hot Water control failed in fan coil unit " + FanCoil( FanCoilNum ).Name,
+										FanCoil( FanCoilNum ).BadMassFlowLimIndexH );
+								}
```

Comment:
Where's the other } else if ( SolFlag == -2 ) ? This looks wrong. See line 2027.